model/camera/camera.go

```diff
@@ -152,6 +154,45 @@ func (c *component) DoCommand(ctx context.Context, command map[string]interface{
 			"command": "fetch",
 			"video":   videoBytesBase64,
 		}, nil
+	case "get-storage-state":
+		c.logger.Debug("get-storage-state command received")
+		state, err := c.videostore.GetStorageState()
+		if err != nil {
+			return nil, err
+		}
+
+		vsConfig := c.videostore.GetConfig()
+		storagePath := vsConfig.Storage.StoragePath
+		storageLimitGB := vsConfig.Storage.SizeGB
+
+		fsUsage, err := diskusage.Statfs(storagePath)
+		var remainingGB float64
+		if err != nil {
+			c.logger.Warnw("failed to get filesystem stats for remaining space", "path", storagePath, "error", err)
+		} else {
+			remainingGB = float64(fsUsage.AvailableBytes) / float64(gigabyte)
+		}
```

Comment:
Have you observed this failing?  I think we should return an error if this happens. Otherwise the caller won't be able to tell the difference between no remaining disk space and remaining disk space, but a failed Statfs call...

Reply:
I have not. I agree this is severe enough to make it an error.

Reply:
Changed to return error on statfs failure